client-tools/Shell/ProcessMonitoringFeature.cpp

```diff
@@ -0,0 +1,158 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Wilfried Goesgens
+////////////////////////////////////////////////////////////////////////////////
+
+#include <chrono>
+
+#include "Basics/Mutex.h"
+#include "Basics/MutexLocker.h"
+#include "Basics/application-exit.h"
+
+#include "Logger/LogMacros.h"
+#include "Logger/Logger.h"
+#include "Logger/LoggerStream.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+
+#include "V8/v8-deadline.h"
+#include "V8/V8SecurityFeature.h"
+
+#include "ProcessMonitoringFeature.h"
+
+using namespace arangodb::basics;
+using namespace arangodb::options;
+using namespace arangodb;
+
+namespace arangodb {
+
+void ProcessMonitoringFeature::addMonitorPID(ExternalId const& pid) {
+  MUTEX_LOCKER(mutexLocker, _MonitoredExternalProcessesLock);
```

Comment:
_[A-Z] is not allowed for standard C++.

It's reserved for standard library.

Also it's not correspond to our codestyle